File: Roadmap/14 - FECHAS/python/raulG91.py
from datetime import datetime as d

today = d.today()
birth_day = d(1991,9,11,15,00,00)
print(today.isoformat(sep=' ',timespec='seconds'))
print(birth_day.isoformat(sep=' ',timespec='seconds'))

diff = today -birth_day
print(diff)

#Extra
print("Extra")
#Format DD-MM-YYYY
print(f'{birth_day.day}-{birth_day.month}-{birth_day.year}')

#Format HH:mm:ss
print(f'{birth_day.hour}:{birth_day.minute}:{birth_day.second}')

#Week of the month
calendar_data = birth_day.isocalendar()
print(f'{calendar_data[1]}')

#Day of the week
day_week = birth_day.weekday()
if day_week == 0:
    day_week ="Monday"
elif day_week == 1:
    day_week = "Tuesday"
elif day_week == 2:
    day_week = "Wednesday"
elif day_week == 3:
    day_week = "Thursday"
elif day_week == 4:
    day_week = "Friday" 
elif day_week == 5:
    day_week = "Satruday" 
elif day_week == 6:
    day_week = "Sunday"

print(day_week)                      

#Month name 
print(birth_day.strftime("%B"))

#Day name + month name
print(birth_day.strftime("%A %d. %B %Y"))

#Extract day of the month 
tuple = birth_day.timetuple()
print(tuple[7])

File: Roadmap/14 - FECHAS/javascript/raulG91.js

let birth_day = new Date(1991,8,11,15,0,0,0);
let now = new Date();
console.log(birth_day);
console.log(now);

let dif = now.getFullYear() - birth_day.getFullYear();
console.log(`Diference in year is ${dif}`);

/*Extra*/

//Day of the week
const formatter_day = new Intl.DateTimeFormat('en', { weekday: 'long' });
day_name = formatter_day.format(birth_day);
console.log(`Day of the week for birth date was: ${day_name}`);

//Name of the month
const formatter = new Intl.DateTimeFormat('en', { month: 'long' });
month_name = formatter.format(birth_day);
console.log(`Month for birth date was ${month_name}`);

//Format date Spanish format
console.log('Formatting date in Spanish version: ',new Intl.DateTimeFormat('es').format(birth_day));

//Format date USA format
console.log('Formatting date in USA version: ',new Intl.DateTimeFormat('en-US').format(birth_day));

// Format date HH:MM:SS
console.log(`Formating HH:MM:SS: ${birth_day.getHours()}:${birth_day.getMinutes()}:${birth_day.getSeconds()}`)
